plugins/experiment_procedure.cc

```diff
@@ -415,73 +478,52 @@ run(const ProcedureRunConfig & run,
+        accuracyConfig.testingData = runProcConf.testingData ? *runProcConf.testingData 
+                                                             : runProcConf.trainingData;
+        accuracyConfig.testingData.stm->where = datasetFold.testing_where;
 
-        ML::Timer timer;
+        // create empty testing procedure
+        if(progress == 0)
+            createAccuracyProcedure(accuracyConfig);
 
-        if(progress == 0) {
-            PolyConfig accuracyProcPC;
-            accuracyProcPC.id = runProcConf.experimentName + "_scorer";
-            accuracyProcPC.type = "classifier.test";
-            accuracyProcPC.params = accuracyConf;
+        auto accuracyOutput = runAccuracyFor(accuracyConfig);
 
-            cerr << " >>>>> Creating testing procedure" << endl;
-            accuracyProc = obtainProcedure(server, accuracyProcPC, onProgress2);
+        // run evaluation on training
+        std::tuple<RunOutput, double> accuracyOutputTrain;
+        if(runProcConf.evalTrain) {
+            accuracyConfig.testingData = runProcConf.trainingData;
+            accuracyConfig.testingData.stm->where = datasetFold.training_where;
 
-            resourcesToDelete.push_back("/v1/procedures/"+accuracyProcPC.id.utf8String());
+            accuracyOutputTrain = runAccuracyFor(accuracyConfig);
         }
 
-        if(!accuracyProc) {
-            throw ML::Exception("Was unable to obtain accuracy procedure");
-        }
-
-        ProcedureRunConfig accuracyProcRunConf;
-        accuracyProcRunConf.id = "run_"+to_string(progress);
-        accuracyProcRunConf.params = jsonEncode(accuracyConf);
-        Date testStart = Date::now();
-        RunOutput accuracyOutput = accuracyProc->run(accuracyProcRunConf, onProgress2);
-        Date testFinish = Date::now();
-
-        cerr << "accuracy took " << timer.elapsed() << endl;
-
-//          cout << jsonEncode(accuracyOutput.results).toStyledString() << endl;
-//          cout << jsonEncode(accuracyOutput.details).toStyledString() << endl;
+        // cout << jsonEncode(accuracyOutput.results).toStyledString() << endl;
+        // cout << jsonEncode(accuracyOutput.details).toStyledString() << endl;
 
         Json::Value duration;
         duration["train"] = trainFinish.secondsSinceEpoch() - trainStart.secondsSinceEpoch();
-        duration["test"]  = testFinish.secondsSinceEpoch() - testStart.secondsSinceEpoch();
+        duration["test"] = get<1>(accuracyOutput) + (runProcConf.evalTrain ? get<1>(accuracyOutputTrain)
+                                                                           : 0);
         durationStatsGen.accumStats(duration, "");
 
         // Add results
         Json::Value foldRez;
         foldRez["fold"] = jsonEncode(datasetFold);
         foldRez["modelFileUrl"] = clsProcConf.modelFileUrl.toUtf8String();
-        foldRez["results"] = jsonEncode(accuracyOutput.results);
+        foldRez["results_test"] = jsonEncode(get<0>(accuracyOutput).results);
```

Comment:
std::move the results? I assume its a container, and we dont seem to need it afterwards?